### PR TITLE
finish intersect function and test in Set structure

### PR DESCRIPTION
**Pull request by gwong5**

### CODE REVIEW

File: src/set.js [hunk 1]
```diff
@@ -1,57 +1,55 @@
...
+        return this.arr.length
+    }
+
+    union(otherSet) {
+        var joined = otherSet.concat(this.arr)
+        return joined
+    }
+
+    intersect(arr2) {
+        var arr3 = []
```
Comment (punitrathore) [inline src/set.js]: please name variables so that they indicate what they are supposed to contain

File: src/set.js [hunk 2]
```diff
@@ -1,57 +1,55 @@
...
+    //   }
+    //   return callback
+    // }
+
+    size() {
+        return this.arr.length
+    }
+
+    union(otherSet) {
+        var joined = otherSet.concat(this.arr)
```
Comment (punitrathore) [inline src/set.js]: This seems incorrect, as the `union` function should return the unique elements.

File: src/set.js [hunk 3]
```diff
@@ -1,57 +1,55 @@
...
+        this.arr.push(element)
+        return this.arr
+    }
+
+    isEmpty() {
+        return this.arr < 1
+    }
+
+    contains(element) {
+        return this.arr.indexOf(element) !== -1 ? true : false
```
Comment (punitrathore) [inline src/set.js]: No need of the ternary operator here. This can be changed to `return this.arr.indexOf(element) !== -1`, since the comparison operator returns a boolean

File: src/set.js (hunk 2 above)
Comment (gwong5) [reply src/set.js]: so I just need to take it one step further and remove duplicates, is this correct?

File: src/set.js [hunk 4]
```diff
@@ -1,57 +1,55 @@
...
-  function intersect(this.arr, this.arr2) {
-    var empty;
-    if (this.arr2.length > this.arr.length) empty = this.arr2, this.arr2 = this.arr, this.arr = empty; // indexOf to loop over shorter
-    return this.arr.filter(function (empty) {
-        if (otherSet.indexOf(empty) !== -1) return true;
-    })
-} return (this.arr, this.arr2)
-
+    constructor() {
+        this.arr = []
```
Comment (deonna) [inline src/set.js]: Avoid using identifiers that contain the name of the data type assigned to them.

Call it something like `this.elements`.

File: src/set.js [hunk 5]
```diff
@@ -1,57 +1,55 @@
...
+    }
+
+    add(element) {
+        this.arr.push(element)
+        return this.arr
+    }
+
+    isEmpty() {
+        return this.arr < 1
```
Comment (deonna) [inline src/set.js]: `this.array.length < 1`

File: src/set.js [hunk 6]
```diff
@@ -1,57 +1,55 @@
...
+        return this.arr.indexOf(element) !== -1 ? true : false
+    }
+
+    remove(element) {
+        var exists = this.arr.indexOf(element)
+        exists !== -1 ? this.arr.splice(exists, 1) : null
+        return this.arr
+    }
+
+    // forEvery() {
```
Comment (deonna) [inline src/set.js]: Remove commented-out lines.

File: src/set.js (hunk 2 above)
Comment (deonna) [reply src/set.js]: To illustrate:

{ 1, 2, 3 } ∪ { 2, 3, 4 } is { 1, 2, 3, 4 }

File: src/set.js (hunk 1 above)
Comment (deonna) [reply src/set.js]: Use `const` if you don't plan on reassigning a variable.

File: src/set.js [hunk 7]
```diff
@@ -1,57 +1,55 @@
...
+    }
+
+    intersect(arr2) {
+        var arr3 = []
+        this.arr.filter(function(element) {
+          if (arr2.indexOf(element) !== -1) {
+            arr3.push(element)
+          }
+        })
+        var removeDuplicates = arr3.filter(function(element, pos) {
```
Comment (deonna) [inline src/set.js]: Call this `withoutDuplicates` or `duplicatesRemoved`. Reserve verbs for method and function names.

File: src/set.js [hunk 8]
```diff
@@ -1,57 +1,55 @@
...
+    isEmpty() {
+        return this.arr < 1
+    }
+
+    contains(element) {
+        return this.arr.indexOf(element) !== -1 ? true : false
+    }
+
+    remove(element) {
+        var exists = this.arr.indexOf(element)
```
Comment (deonna) [inline src/set.js]: `const`

The ternary makes this code hard to parse.

```javascript
const exists = this.arr.indexOf(element)

if (exists !== -1) {
    this.arr.splice(exists, 1)
    return this.arr
}

```

File: spec/set.js [hunk 9]
```diff
@@ -65,15 +65,15 @@ describe('Set', () => {
...
-    context('forEvery()', () => {
-      it('takes a callback function and passes it each element in sequence.', () => {
-        const mySet = new Set()
-        mySet.add('bar')
-        mySet.add('is')
-        // mySet.add('foo')
-        expect(mySet.forEvery()).to.eql(undefined)
-      })
-    })
+    // context('forEvery()', () => {
```
Comment (deonna) [inline spec/set.js]: Remove commented-out lines.

**Review by deonna — COMMENTED** [review]
See individual comments.
